spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/eureka/DiscoveryClientConfiguration.java

```diff
@@ -83,8 +82,7 @@ public EurekaClient eurekaClient(ApplicationInfoManager applicationInfoManager,
 		@SneakyThrows
 		@org.springframework.cloud.context.config.annotation.RefreshScope
 		public EurekaClient eurekaClient(ApplicationInfoManager applicationInfoManager,
-										 EurekaClientConfig config, EurekaInstanceConfig instance) {
-			applicationInfoManager.initComponent(instance);
```

Comment:
@dsyer `applicationInfoManager.initComponent(instance)` is deprecated and was resetting the `InstanceInfo` (with port = 0). What does this fix? I'm assuming it doesn't reset when you call `/refresh`? Eureka config first works with this change.